applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -27,9 +29,26 @@
 except ImportError:
     convdiff_response_factory = None
 
-
 import time as timer
 
+class IterationScope:
+    def __init__(self, output_folder_name, response_id, iteration_number):
+        if (output_folder_name != None):
+            self.currentPath = pathlib.Path.cwd()
+            output_path = pathlib.Path(output_folder_name)
+            response_text = "response_{:}/{:d}".format(response_id, iteration_number)
+            self.scope = output_path / response_text
+
+    def __enter__(self):
+        if (hasattr(self, "scope")):

```

Comment:
i think this check is not needed, since you set this variable in the constructor?

Reply:
not exactly. I only set the scope if folder structure is required only.

Reply:
I made it like that to have minimal cost in the case if this folder structure is not required.

Reply:
removed `hasattr`